pkg/plugins/connection/ssh.go

```diff
@@ -40,13 +45,59 @@ func (t *SSHOpen) Metadata() *gornir.TaskMetadata {
 	return t.Meta
 }
 
+type authMethodResult struct {
+	authMethods []ssh.AuthMethod
+	err         error
+}
+
+func getAuthMethods(host *gornir.Host, logger gornir.Logger) (*[]ssh.AuthMethod, error) {
+	var authMethods = []ssh.AuthMethod{ssh.Password(host.Password)}
+	// GetStrict should return a default value per `man ssh_config` if this fails, it is because we couldn't parse the config file
+	sshPrivKeyFname, err := ssh_config.GetStrict(host.Hostname, "IdentityFile")
+	if err != nil {
+		return nil, err
+	}
+	signer, err := getPubKeySigner(host, sshPrivKeyFname, logger)
+	// Drop private key auth from the list and fallback to user/pass
+	if err != nil {
```

Comment:
I still don't like this but I get why you are doing it. See general comment.

Reply:
The specific problem with private key auth is that a default value is returned for the private key file per `man ssh_config`.
```
     IdentityFile
             Specifies a file from which the user's DSA, ECDSA, Ed25519 or RSA authentication identity is read.  The default is ~/.ssh/id_dsa, ~/.ssh/id_ecdsa, ~/.ssh/id_ed25519 and ~/.ssh/id_rsa.  Additionally, any identities represented by the authentication agent
             will be used for authentication unless IdentitiesOnly is set.  If no certificates have been explicitly specified by CertificateFile, ssh(1) will try to load certificate information from the filename obtained by appending -cert.pub to the path of a specified
             IdentityFile.
```
For example, on my system, if I move my `~/.ssh/config`, I get the following error:
```
unable to read private key: open /Users/djustice/.ssh/identity: no such file or directory
```
Not everyone will have a valid private key setup, and I don't want to fail the connection because of a missing default key.  What do you think, does that make sense?